src/interfaces/wallet.cpp

```diff
@@ -457,7 +457,7 @@ class WalletImpl : public Wallet
         return result;
     }
     unsigned int getConfirmTarget() override { return m_wallet.m_confirm_target; }
-    bool hdEnabled() override { return m_wallet.IsHDEnabled(); }
+    bool hdEnabled() override { return m_wallet.HasHDSeed(); }
```

Comment:
Let's fix `hdEnabled` in `WalletModel` instead, otherwise things are going to get confusing fast.

Reply:
I've changed it so that `hdEnabled()` does the same thing as before and instead added a `hasHDSeed()` which calls `HasHDSeed`. `hasHDSeed` is used instead of `hdEnabled` in the `walletmodel.cpp` changes.